src/databricks/labs/ucx/source_code/files.py

```diff
@@ -14,47 +19,45 @@
 logger = logging.getLogger(__name__)
 
 
-class SourceFile(SourceContainer):
+class LocalFile(SourceContainer):
 
     def __init__(self, path: str, source: str, language: Language):
         self._path = path
         self._original_code = source
         # using CellLanguage so we can reuse the facilities it provides
         self._language = CellLanguage.of_language(language)
 
-    @property
-    def object_type(self) -> ObjectType:
-        return ObjectType.FILE
-
-    def build_dependency_graph(self, graph: DependencyGraph) -> None:
+    def build_dependency_graph(self, parent: DependencyGraph) -> None:
         if self._language is not CellLanguage.PYTHON:
             logger.warning(f"Unsupported language: {self._language.language}")
+            return
         linter = ASTLinter.parse(self._original_code)
         run_notebook_calls = PythonLinter.list_dbutils_notebook_run_calls(linter)
         notebook_paths = {PythonLinter.get_dbutils_notebook_run_path_arg(call) for call in run_notebook_calls}
         for path in notebook_paths:
-            graph.register_dependency(Dependency(ObjectType.NOTEBOOK, path))
+            parent.register_dependency(UnresolvedDependency(path))
         # TODO https://github.com/databrickslabs/ucx/issues/1287
         import_names = PythonLinter.list_import_sources(linter)
         for import_name in import_names:
-            # we don't know yet if it's a file or a library
-            graph.register_dependency(Dependency(None, import_name))
-
-
-class WorkspaceFile(SourceFile):
-    pass
-
-
-class LocalFile(SourceFile):
-    pass
+            parent.register_dependency(UnresolvedDependency(import_name))
 
 
 class LocalFileMigrator:
-    """The Files class is responsible for fixing code files based on their language."""
+    """The LocalFileMigrator class is responsible for fixing code files based on their language."""
 
-    def __init__(self, languages: Languages):
+    def __init__(self, languages: Languages, resolver: DependencyResolver):
         self._languages = languages
         self._extensions = {".py": Language.PYTHON, ".sql": Language.SQL}
+        self._resolver = resolver
+
+    def build_dependency_graph(self, path: Path) -> DependencyGraph:
+        dependency = self._resolver.resolve_dependency(UnresolvedDependency(str(path)))
```

Comment:
```suggestion
        dependency = self._resolver.resolve_dependency(UnresolvedDependency(path))
```
use `pathlib.Path` everywhere for paths.